src/components/DoNotSellAcknowledgement.tsx

```diff
@@ -0,0 +1,79 @@
+// external
+import { h, JSX } from 'preact';
+import { useIntl } from 'react-intl';
+
+// global
+import { noticeAndDoNotSellMessages } from '../messages';
+import type { HandleSetViewState, NavigatorWithGPC } from '../types';
+import { useAirgap } from '../hooks';
+
+// local
+import Button from './Button';
+import ColumnContent from './ColumnContent';
+import Title from './Title';
+import { Purpose } from '@transcend-io/airgap.js-types';
+import { useEffect } from 'preact/hooks';
+
+/**
+ * Component showing acknowledgement of do not sell
+ */
+export default function DoNotSellAcknowledgement({
+  handleSetViewState,
+}: {
+  /** Function to change viewState */
+  handleSetViewState: HandleSetViewState;
+}): JSX.Element {
+  const { airgap } = useAirgap();
+  const { formatMessage } = useIntl();
+
+  const handleOptOut: JSX.MouseEventHandler<HTMLButtonElement> | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
+    // Confirm current consent
+    airgap.setConsent(event, {
+      ...airgap.getConsent().purposes,
+      [Purpose.SaleOfInfo]: false,
+    });
+    console.log(airgap.getConsent());
```

Comment:
Any call to `setConsent()` will confirm consent (persist existing values & set `.confirmed = true`). Also, you can modify individual keys in `setConsent()`.

```suggestion
    // Confirm current consent
    airgap.setConsent(event, {
      [Purpose.SaleOfInfo]: false,
    });
```